cpp/src/arrow/compute/exec/util.h

```diff
@@ -36,7 +36,6 @@
 #include "arrow/util/logging.h"
 #include "arrow/util/mutex.h"
 #include "arrow/util/thread_pool.h"
-#include "arrow/util/tracing_internal.h"
```

Comment:
Note, this header is still an internal header as it includes `arrow/util/logging.h` which is perhaps our most egregious improperly named header.

Reply:
Though this won't break an external project build, right?

Reply:
Ah, I don't know.  But we have a check that ensures we don't expose `logging.h` from `api.h`.  So we need to keep `map_node.h` out of `api.h` until this is resolved.

Reply:
Done.